### PR TITLE
feat: pathwar compose up --force-recreate

### PR DESCRIPTION
**Pull request by Z-a-r-a-k-i**

### CODE REVIEW

File: go/pkg/pwcompose/compose.go [hunk 1]
```diff
@@ -222,13 +227,19 @@ func Up(preparedCompose string, instanceKey string, logger *zap.Logger) error {
 	}
 	tmpFile.Close()
 
+	// down instances if force recreate
+	if forceRecreate {
+		Down(ctx, []string {challengeId}, false, false, cli, logger)
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]: err check?

File: go/pkg/pwcompose/compose.go [hunk 2]
```diff
@@ -169,12 +169,15 @@ func Prepare(challengeDir string, prefix string, noPush bool, version string, lo
...
-	logger.Debug(
-		"up",
-		zap.String("compose", preparedCompose),
-		zap.String("instance-key", instanceKey),
-	)
+func Up(
+	ctx context.Context,
+	preparedCompose string,
+	instanceKey string,
+	forceRecreate bool,
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]: if you prefer, you can put all the non-necessary vars in a new `UpOpts struct` (`instanceKey`, `forceRecreate`)

File: go/pkg/pwcompose/compose.go (hunk 1 above)
Comment (Z-a-r-a-k-i) [reply go/pkg/pwcompose/compose.go]: oupss

File: go/pkg/pwcompose/compose.go (hunk 2 above)
Comment (Z-a-r-a-k-i) [reply go/pkg/pwcompose/compose.go]: will do only if we have more than 2 args

Comment (moul) [reply go/pkg/pwcompose/compose.go]: same for me

File: go/pkg/pwcompose/compose.go [hunk 3]
```diff
@@ -222,13 +227,22 @@ func Up(preparedCompose string, instanceKey string, logger *zap.Logger) error {
...
+	}
+
 	// start instances
 	logger.Debug("docker-compose", zap.String("action", "up"))
 	cmd := exec.Command("docker-compose", "-f", tmpPreparedComposePath, "up", "-d")
 	cmd.Stdout = os.Stderr
 	cmd.Stderr = os.Stderr
 	err = cmd.Run()
 	if err != nil {
+		fmt.Println("Error detected while starting containers, it's probably due to a conflict with previously created containers that share the same name. You should retry with --force-recreate flag")
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]: use the logger

Comment (Z-a-r-a-k-i) [reply go/pkg/pwcompose/compose.go]: done

File: go/pkg/pwcompose/compose.go [hunk 4]
```diff
@@ -222,13 +227,22 @@ func Up(preparedCompose string, instanceKey string, logger *zap.Logger) error {
...
+	}
+
 	// start instances
 	logger.Debug("docker-compose", zap.String("action", "up"))
 	cmd := exec.Command("docker-compose", "-f", tmpPreparedComposePath, "up", "-d")
 	cmd.Stdout = os.Stderr
 	cmd.Stderr = os.Stderr
 	err = cmd.Run()
 	if err != nil {
+		logger.Error("Error detected while starting containers, it's probably due to a conflict with previously created containers that share the same name. You should retry with --force-recreate flag")
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]:
```suggestion
		logger.Error("Error detected while starting containers, it's probably due to a conflict with previously created containers that share the same name. You should retry with --force-recreate flag", zap.Error(err))
```

**Pull request merged**